apps/shipments/admin.py

```diff
@@ -123,4 +135,208 @@ def has_change_permission(self, request, obj=None):
         return False
 
 
+class BaseModelHistory(SimpleHistoryAdmin):
```

Comment:
This admin class allow modification to the Shipment instance via the History view.  This should _not_ be an updatable view for a Shipment.  The entirety of the admin.py class after line 138 can be replace with the following code.  This will provide us with a consistent Admin view for the current Shipment as well as all its history.  With both views being read-only as desired.

```Python

class HistoricalShipmentAdmin(SimpleHistoryAdmin, ShipmentAdmin):
    readonly_fields = [field.name for field in Shipment._meta.get_fields()]

    def has_change_permission(self, request, obj=None):
        return True


admin.site.register(Shipment, HistoricalShipmentAdmin)
```

Reply:
# In `history_view`:
When trying to do what Lucas suggested, I got the following error:
```suggession
 FieldError: Non-relational field given in select_related: 'history_user'.
```
Due to the following portion of code that I ended up removing:
```suggession
 if not isinstance(history.model.history_user, property):
   action_list = action_list.select_related("history_user")
```
Since we customized `historical_user` to be a `CharField` instead of a `ForeignKey` as the library expects

In that same view since the page is read only, I got rid of this portion of code:
```suggession
 if not self.has_change_permission(request, obj):
   raise PermissionDenied
```

# In `history_form_view`:
This portion of code has been removed for the same reasons as above:
```suggession
 if not self.has_change_permission(request, obj):
   raise PermissionDenied
```
I also did Simplify this portion of code from:
```suggession
 if SIMPLE_HISTORY_EDIT:
   change_history = True
 else:
   change_history = False
```
To:
`change_history = bool(SIMPLE_HISTORY_EDIT)`

which can further being simplified to:
`change_history = settings.SIMPLE_HISTORY_EDIT`

Since changes are not allowed the following code has been removed:
```suggession
 if request.method == "POST":
   form = form_class(request.POST, request.FILES, instance=obj)
   if form.is_valid():
       new_object = self.save_form(request, form, change=True)
       self.save_model(request, new_object, form, change=True)
       form.save_m2m()
       self.log_change(
           request,
           new_object,
           self.construct_change_message(request, form, formsets),
       )
       return self.response_change(request, new_object)
```

Lastly, changed the context parameter: `change` value  to be `False`

Reply:
Okay, it makes sense why you needed to change those functions. I still don't like having copied code from a dependency - ideally we would fork the dependency and make the changes in the fork. Otherwise it is going to be a maintenance nightmare when `django-simple-history` has changes. But this is acceptable for now.